src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -381,5 +387,54 @@ public static string CreatePropertyModelName(string prefix, string propertyName)
                 return prefix + "." + propertyName;
             }
         }
+
+        internal static void ClearModelStateDictionaryForModel(
+            [NotNull] Type modelType,
+            [NotNull] ModelStateDictionary modelstate,
+            [NotNull] IModelMetadataProvider metadataProvider,
+            string modelKey)
+        {
+            if (string.IsNullOrEmpty(modelKey))
+            {
+                var modelMetadata = metadataProvider.GetMetadataForType(modelType);
+                if (modelMetadata.IsCollectionType)
+                {
+                    var elementType = GetElementType(modelMetadata.ModelType);
+                    modelMetadata = metadataProvider.GetMetadataForType(elementType);
+                }
+
+                foreach (var property in modelMetadata.Properties)
+                {
+                    var propertyBindingName = property.BinderModelName ?? property.PropertyName;
+                    var childKey = ModelBindingHelper.CreatePropertyModelName(modelKey, propertyBindingName);
```

Comment:
Consider making an overload of `CreatePropertyName` taking a `ModelMetadata`. The `property.BinderModelName ?? property.PropertyName` logic is repeated all over the place.